packages/adblocker-benchmarks/blockers/tsurlfilter.js

```diff
@@ -0,0 +1,55 @@
+/*!
+ * Copyright (c) 2017-present Cliqz GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+const {
+  Engine,
+  Request,
+  RequestType,
+  RuleStorage,
+  StringRuleList,
+  setConfiguration,
+} = require('@adguard/tsurlfilter/dist/bundle.min.cjs');
+
+const requestTypes = new Map([
+  [ 'sub_frame', RequestType.Subdocument ],
+  [ 'script', RequestType.Script ],
+  [ 'stylesheet', RequestType.Stylesheet ],
+  [ 'image', RequestType.Image ],
+  [ 'xmlhttprequest', RequestType.XmlHttpRequest ],
+  [ 'media', RequestType.Media ],
+  [ 'font', RequestType.Font ],
+  [ 'other', RequestType.Other ],
+]);
+
+module.exports = class TSUrlFilter {
+  static async initialize() {
+    setConfiguration({
+      engine: 'extension',
+      version: '1.0.0',
+      verbose: true,
+    });
+  }
+
+  static async parse(rawLists) {
+    const list = new StringRuleList(1, rawLists, true, true, true);
+    const ruleStorage = new RuleStorage([ list ]);
+    return new TSUrlFilter(new Engine(ruleStorage));
+  }
+
+  constructor(engine) {
+    this.engine = engine;
+  }
+
+  match({ url, frameUrl, type }) {
+    const requestType = requestTypes.get(type) || RequestType.Other;
+    const request = new Request(url, frameUrl, requestType);
+    const result = this.engine.matchRequest(request);
```

Comment:
@ameshkov I notice that the filter matching is quite slow. In the API, there's a function for returning _all_ matches and then another function for returning only the basic match. Ideally there should be one function for returning the basic match, and the search should be terminated when a match is found. I hope I'm not using the wrong function here. Let me know if there's already a better way to do this.

Reply:
@mjethani No worries, it's still a work in progress at this point, we're only entering the performance tuning stage.